src/Microsoft.Extensions.DependencyInjection/ServiceProvider.cs

```diff
@@ -186,7 +186,7 @@ public void Dispose()
             }
         }
 
-        internal object CaptureDisposable(object service)
+        internal virtual object CaptureDisposable(object service)
```

Comment:
Having performance tests is fine but I think having a way to verify the call isn't actually made is just as valuable (if not more).